air-quality-ui/src/components/summary-grid/cell/cell-rules/ColourCell.ts

```diff
@@ -0,0 +1,53 @@
+export type PollutantData = {
+  [key: string]: {
+    aqiLevel: number
+    value: number
+  }
+}
+
+export type Params = {
+  column: { colId: string }
+  value: number | undefined
+  data: {
+    aqiDifference: string
+    forecast: PollutantData
+    measurements: PollutantData | undefined
+  }
+}
+
+export default function colourCell(
+  showAllColoured: boolean,
+  params: Params,
+  lowerLimit: number,
+  upperLimit?: number,
+): boolean {
+  const pollutantType: string = params.column.colId.split('.')[1]
+  if (!params.value) return false
+
+  if (upperLimit === undefined && showAllColoured) {
+    return params.value >= lowerLimit
+  }
+
+  let isInColourBand
+  if (upperLimit === undefined) {
+    isInColourBand = params.value >= lowerLimit
+  } else {
+    isInColourBand = params.value >= lowerLimit && params.value <= upperLimit
+  }
+
+  if (showAllColoured) return isInColourBand
+  if (!params.data.measurements || !params.data.measurements[pollutantType])
+    return false
+
+  const aqiDifference = Math.abs(
+    params.data.forecast[pollutantType].aqiLevel -
```

Comment:
Just to check, are these the overall aqiLevels that show in the summary, i.e. the columns on the far left?

Reply:
No these are the aqi levels for each polutant. : )